### PR TITLE
[win] migrate windows tests to civ2

### PR DESCRIPTION
**Pull request by aslonnie**
and simplify test setup a bit:

one difference: it checks out outside container, and mount in the dir; needs to perform another clone for the symlink issues, but enables including the env setup inside the source code rather than inlining them on the pipeline steps.

### CODE REVIEW

**Comment by aslonnie** [comment]
tests passed.

some runtime env related tests are flaky though. @rickyyx 

I am particularly not very happy about the `test_invalid_conda_env` one:

````
# Sleep to wait bad runtime env cache removed.
--
  | time.sleep(bad_runtime_env_cache_ttl_seconds)
  |  
  | # The third time this runs it should be slower as the error isn't cached.
  | start = time.time()
  | with pytest.raises(RuntimeEnvSetupError, match="ResolvePackageNotFound"):
  | ray.get(f.options(runtime_env=bad_env).remote())
  |  
  | >       assert (time.time() - start) > (first_time / 2.0)
  | E       assert (1700085106.696519 - 1700085092.9776516) > (28.069849491119385 / 2.0)
  | E        +  where 1700085106.696519 = <built-in function time>()
  | E        +    where <built-in function time> = time.time

````

File: .buildkite/windows.rayci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+group: ":windows: windows"
+steps:
+- label: ":windows: python tests"
```
Comment (can-anyscale) [inline .buildkite/windows.rayci.yml:3]: can we move these to individual team rayci (core, data, serve, ml, etc.)

File: ci/ci.sh [hunk 2]
```diff
@@ -292,6 +291,33 @@ test_python() {
   fi
 }
 
+# For running Python tests on Windows (excluding Serve).
+test_train_windows() {
```
Comment (can-anyscale) [inline ci/ci.sh:295]: can we add a flag `is_flaky` in these functions and have a flaky list vs. non-flaky list of targets; then run the flaky ones with soft_fail?

**Review by can-anyscale — COMMENTED** [review]
Overall looking good. The total machine time is 2X compared to before though so it's difficult to tell if this will incur more or less cost.

File: ci/ci.sh (hunk 2 above)
Comment (aslonnie) [reply ci/ci.sh:295]: maybe in a follow up? I can even try use your `rayci` wrapper.

File: .buildkite/windows.rayci.yml (hunk 1 above)
Comment (aslonnie) [reply .buildkite/windows.rayci.yml]: sure. moved.

in v1, all windows tests are in one group. hope that people can still find them.

File: ci/ci.sh (hunk 2 above)
Comment (aslonnie) [reply ci/ci.sh:295]: these are just a copy of the `test_python` function, so that ml train tests can have their own target.

**Comment by aslonnie** [comment]
> Overall looking good. The total machine time is 2X compared to before though so it's difficult to tell if this will incur more or less cost.

will optimize, will optimize it.

File: ci/ci.sh (hunk 2 above)
Comment (can-anyscale) [reply ci/ci.sh:295]: sound good

**Review by can-anyscale — APPROVED** [review]

**Comment by aslonnie** [comment]
@peytondmurray and @mattip , please review and comment, or will merge this at the end of the day.

**Comment by aslonnie** [comment]
any feedback?

**Comment by aslonnie** [comment]
merging in 30min.

**Pull request merged**